graphql_compiler/schema_generation/sqlalchemy/schema_graph_builder.py

```diff
@@ -126,10 +126,21 @@ def _get_edge_types(tables):
     edge_types = []
     for table in six.itervalues(tables):
         for column in table.get_children():
-            for foreign_key in column.foreign_keys:
+            if len(column.foreign_keys) > 1:
+                raise IllegalSchemaInputError(u'Multiple foreign keys are defined on column {} of '
+                                              u'table {}.'.format(column.key, table.name))
+            if len(column.foreign_keys) > 0:
+                foreign_key = list(column.foreign_keys)[0]
+
+                outgoing_table = foreign_key.column.table
+                if not foreign_key.column.primary_key:
+                    raise AssertionError(u'Foreign key defined in column {} in table {} references '
+                                         u'column {} in table {}, which is not primary key.'
+                                         .format(table.name, column.key, foreign_key.column.name,
+                                                 outgoing_table.name))
+
                 # We prepend the table name to ensure that the name is unique.
                 edge_name = '{}_{}'.format(table.name, column.name)
```

Comment:
Could this result in collisions, thinking adversarially?

Reply:
Yes it could. I guess that we must both preprend and append the column name to ensure that the schema element is unique.

Reply:
Also, how do we handle edges that result from join tables? This is a very complex design question, please provide an overview of your proposed design in the PR summary.

Reply:
Yea not that I think about it prohibiting tables with  '_' is a terrible idea.

Reply:
A strictly better idea is detecting and prohibiting precisely schemas that collide in this way. I don't expect it to happen at all IRL. But I'm also not saying this is the best solution.